### PR TITLE
Redeliver configupdate fat bucket

### PR DESCRIPTION
**Pull request by KyleAure**
Previous PR reverted: #7802 
Using a new branch to work on this test bucket

### CODE REVIEW

**Comment by KyleAure** [comment]
#build

File: dev/com.ibm.ws.concurrent.persistent_fat_configupdate/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+/bootstrapping.properties
```
Comment (aguibert) [inline dev/com.ibm.ws.concurrent.persistent_fat_configupdate/.gitignore]: this file can be removed

File: dev/com.ibm.ws.concurrent.persistent_fat_configupdate/fat/src/com/ibm/ws/concurrent/persistent/fat/configupd/FATSuite.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import componenttest.topology.impl.LibertyServer;
+import componenttest.topology.impl.LibertyServerFactory;
+
+@RunWith(Suite.class)
+@SuiteClasses({ PersistentExecutorConfigUpdateTest.class })
+public class FATSuite {
+    static LibertyServer server = LibertyServerFactory.getLibertyServer("com.ibm.ws.concurrent.persistent.fat.configupdate");
+
+    /**
+     * Copies the simulated GA repository local files (saved during compilation) to the server install root.
```
Comment (aguibert) [inline dev/com.ibm.ws.concurrent.persistent_fat_configupdate/fat/src/com/ibm/ws/concurrent/persistent/fat/configupd/FATSuite.java]: this block of javadoc is incorrect and can be removed

File: dev/com.ibm.ws.concurrent.persistent_fat_configupdate/publish/.gitignore [hunk 3]
```diff
@@ -0,0 +1,2 @@
+/files
```
Comment (aguibert) [inline dev/com.ibm.ws.concurrent.persistent_fat_configupdate/publish/.gitignore]: this file can be removed

File: dev/com.ibm.ws.concurrent.persistent_fat_configupdate/publish/servers/com.ibm.ws.concurrent.persistent.fat.configupdate/.gitignore [hunk 4]
```diff
@@ -0,0 +1,2 @@
+/derby
```
Comment (aguibert) [inline dev/com.ibm.ws.concurrent.persistent_fat_configupdate/publish/servers/com.ibm.ws.concurrent.persistent.fat.configupdate/.gitignore]: this file can be removed

File: dev/com.ibm.ws.concurrent.persistent_fat_configupdate/publish/servers/com.ibm.ws.concurrent.persistent.fat.configupdate/bootstrap.properties [hunk 5]
```diff
@@ -0,0 +1,6 @@
+bootstrap.include=../testports.properties
```
Comment (aguibert) [inline dev/com.ibm.ws.concurrent.persistent_fat_configupdate/publish/servers/com.ibm.ws.concurrent.persistent.fat.configupdate/bootstrap.properties]: bootstrapping.properties files typically have copyrights in them too, you can use this:
```
###############################################################################
# Copyright (c) 2019 IBM Corporation and others.
# All rights reserved. This program and the accompanying materials
# are made available under the terms of the Eclipse Public License v1.0
# which accompanies this distribution, and is available at
# http://www.eclipse.org/legal/epl-v10.html
#
# Contributors:
#     IBM Corporation - initial API and implementation
###############################################################################```

File: dev/com.ibm.ws.concurrent.persistent_fat_configupdate/publish/servers/com.ibm.ws.concurrent.persistent.fat.configupdate/server.xml [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+        IBM Corporation - initial API and implementation
+ -->
+<server>
+  <featureManager>
+    <feature>concurrent-1.0</feature>
+    <feature>jndi-1.0</feature>
+    <!-- <feature>osgiConsole-1.0</feature> -->
+    <feature>persistentExecutor-1.0</feature>
+    <feature>servlet-3.1</feature>
+    <feature>jdbc-4.2</feature>
```
Comment (aguibert) [inline dev/com.ibm.ws.concurrent.persistent_fat_configupdate/publish/servers/com.ibm.ws.concurrent.persistent.fat.configupdate/server.xml]: it looks like this bucket is set up to run on any Java level, including java 7, but jdbc-4.2 only works for Java 8+. I believe it is valid for these tests to be running on Java 7, so lets downgrade to jdbc-4.1 here.

File: dev/com.ibm.ws.concurrent.persistent_fat_configupdate/test-applications/persistcfgtest/resources/WEB-INF/web.xml [hunk 7]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (aguibert) [inline dev/com.ibm.ws.concurrent.persistent_fat_configupdate/test-applications/persistcfgtest/resources/WEB-INF/web.xml]: this file can be removed

File: dev/com.ibm.ws.concurrent.persistent_fat_configupdate/fat/src/com/ibm/ws/concurrent/persistent/fat/configupd/PersistentExecutorConfigUpdateTest.java [hunk 8]
```diff
@@ -0,0 +1,403 @@
...
+import com.ibm.websphere.simplicity.config.ServerConfiguration;
+import com.ibm.websphere.simplicity.log.Log;
+
+import componenttest.annotation.ExpectedFFDC;
+import componenttest.topology.impl.LibertyServer;
+
+/**
+ * Tests for configuration updates to persistent scheduled executors
+ */
+public class PersistentExecutorConfigUpdateTest {
```
Comment (aguibert) [inline dev/com.ibm.ws.concurrent.persistent_fat_configupdate/fat/src/com/ibm/ws/concurrent/persistent/fat/configupd/PersistentExecutorConfigUpdateTest.java]: for test classes we need to add the `@RunWith(FATRunner.class)` annotation to all test classes (not `FATSuite`s though). This was added automatically at compile-time in CL, but in OL we need to add the annotation manually. The benefit is that it gives us some extra error checking such as FFDC detection and a server cleanup safety-net

File: dev/com.ibm.ws.concurrent.persistent_fat_configupdate/fat/src/com/ibm/ws/concurrent/persistent/fat/configupd/FATSuite.java [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+import org.junit.runners.Suite.SuiteClasses;
+
+import com.ibm.websphere.simplicity.Machine;
+
+import componenttest.topology.impl.LibertyFileManager;
+import componenttest.topology.impl.LibertyServer;
+import componenttest.topology.impl.LibertyServerFactory;
+
+@RunWith(Suite.class)
+@SuiteClasses({ PersistentExecutorConfigUpdateTest.class })
```
Comment (aguibert) [inline dev/com.ibm.ws.concurrent.persistent_fat_configupdate/fat/src/com/ibm/ws/concurrent/persistent/fat/configupd/FATSuite.java]: how long does this config update test take to run? If it takes >5m, we should move some/all of the tests to FULL mode. (If all tests are moved to FULL, remember that a dummy AlwaysPassesTest needs to be added)

**Comment by KyleAure** [comment]
#build

**Pull request merged**